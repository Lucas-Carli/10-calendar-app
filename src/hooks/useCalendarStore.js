import { useDispatch, useSelector } from 'react-redux';
import { onAddNewEvent, onDeleteEvent, onSetActiveEvent, onUpdateEvent } from '../store/calendar/calendarSlice';

export const useCalendarStore = () => {

    const dispatch = useDispatch();
    const { events, activeEvent } = useSelector(state => state.calendar);

    const setActiveEvent = (calendarEvent) => {
        dispatch(onSetActiveEvent(calendarEvent))
    }

    const startSavingEvent = async (calendarEvent) => {
        // TODO : llegar al backend

        // Todo bien
        if (calendarEvent._id) {
            dispatch(onUpdateEvent({ ...calendarEvent }));
        } else {
            // Creando
            dispatch(onAddNewEvent({ ...calendarEvent, _id: new Date().getTime() }))
        }
    }

    const startDeletingEvent = () => {
        // TODO: llegar al backend
        dispatch(onDeleteEvent());
    }


    return {
        //* Properties
        activeEvent,
        events,
        hasEventSelected: !!activeEvent,

        //* Events
        startDeletingEvent,
        setActiveEvent,
        startSavingEvent,
        

    }
}
