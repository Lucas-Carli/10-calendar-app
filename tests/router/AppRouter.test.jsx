import { render, screen } from "@testing-library/react";
import { useAuthStore } from "../../src/hooks/useAuthStore";
import { AppRouter } from "../../src/router/AppRouter";
import { MemoryRouter } from "react-router-dom";
import { CalendarPage } from "../../src/calendar";

jest.mock('../../src/hooks/useAuthStore');

jest.mock('../../src/calendar', () => ({
    CalendarPage: () => <h1>CalendarPage</h1>
}));


describe('Tests en <AppRouter />', () => {

    const mockCheckAuthToken = jest.fn();

    beforeEach(() => jest.clearAllMocks());

    test('must display the loading screen and call the checkAuthToken', () => {

        useAuthStore.mockReturnValue({
            status: 'checking',
            checkAuthToken: mockCheckAuthToken
        });

        render(<AppRouter />);
        expect(screen.getByText('Loading...')).toBeTruthy()
        expect(mockCheckAuthToken).toHaveBeenCalled();

    });

    test('must display login in case of not being authenticated', () => {

        useAuthStore.mockReturnValue({
            status: 'not-authenticated',
            checkAuthToken: mockCheckAuthToken
        });

        const { container } = render(
            <MemoryRouter initialEntries={['/auth2/something/otherthing']} future={{ v7_startTransition: true }}>
                <AppRouter />
            </MemoryRouter>
        );

        // screen.debug();
        expect(screen.getByText('Entry')).toBeTruthy();
        expect(container).toMatchSnapshot();

    });

    test('must show the calendar if we are authenticated', () => {
        useAuthStore.mockReturnValue({
            status: 'authenticated',
            checkAuthToken: mockCheckAuthToken
        });

        render(
            <MemoryRouter initialEntries={['/auth2/something/otherthing']} future={{ v7_startTransition: true }}>
                <AppRouter />
            </MemoryRouter>
        );

        expect(screen.getByText('CalendarPage')).toBeTruthy();

    })

});