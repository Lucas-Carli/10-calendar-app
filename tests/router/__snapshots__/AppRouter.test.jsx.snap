// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests en <AppRouter /> must display calendar if we are authenticated 1`] = `
<div>
  <div
    class="container login-container"
  >
    <div
      class="row"
    >
      <div
        class="col-md-6 login-form-1"
      >
        <h3>
          Entry
        </h3>
        <form>
          <div
            class="form-group mb-2"
          >
            <input
              class="form-control"
              name="loginEmail"
              placeholder="Email"
              type="text"
              value=""
            />
          </div>
          <div
            class="form-group mb-2"
          >
            <input
              class="form-control"
              name="loginPassword"
              placeholder="Password"
              type="password"
              value=""
            />
          </div>
          <div
            class="d-grid gap-2"
          >
            <input
              class="btnSubmit"
              type="submit"
              value="Login"
            />
          </div>
        </form>
      </div>
      <div
        class="col-md-6 login-form-2"
      >
        <h3>
          Register
        </h3>
        <form>
          <div
            class="form-group mb-2"
          >
            <input
              class="form-control"
              name="registerName"
              placeholder="Name"
              type="text"
              value=""
            />
          </div>
          <div
            class="form-group mb-2"
          >
            <input
              class="form-control"
              name="registerEmail"
              placeholder="Email"
              type="email"
              value=""
            />
          </div>
          <div
            class="form-group mb-2"
          >
            <input
              class="form-control"
              name="registerPassword"
              placeholder="Password"
              type="password"
              value=""
            />
          </div>
          <div
            class="form-group mb-2"
          >
            <input
              class="form-control"
              name="registerPassword2"
              placeholder="Repeat the password"
              type="password"
              value=""
            />
          </div>
          <div
            class="d-grid gap-2"
          >
            <input
              class="btnSubmit"
              type="submit"
              value="Create account"
            />
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;

exports[`Tests en <AppRouter /> must display calendar if we have authenticated 1`] = `
<div>
  <div
    class="container login-container"
  >
    <div
      class="row"
    >
      <div
        class="col-md-6 login-form-1"
      >
        <h3>
          Entry
        </h3>
        <form>
          <div
            class="form-group mb-2"
          >
            <input
              class="form-control"
              name="loginEmail"
              placeholder="Email"
              type="text"
              value=""
            />
          </div>
          <div
            class="form-group mb-2"
          >
            <input
              class="form-control"
              name="loginPassword"
              placeholder="Password"
              type="password"
              value=""
            />
          </div>
          <div
            class="d-grid gap-2"
          >
            <input
              class="btnSubmit"
              type="submit"
              value="Login"
            />
          </div>
        </form>
      </div>
      <div
        class="col-md-6 login-form-2"
      >
        <h3>
          Register
        </h3>
        <form>
          <div
            class="form-group mb-2"
          >
            <input
              class="form-control"
              name="registerName"
              placeholder="Name"
              type="text"
              value=""
            />
          </div>
          <div
            class="form-group mb-2"
          >
            <input
              class="form-control"
              name="registerEmail"
              placeholder="Email"
              type="email"
              value=""
            />
          </div>
          <div
            class="form-group mb-2"
          >
            <input
              class="form-control"
              name="registerPassword"
              placeholder="Password"
              type="password"
              value=""
            />
          </div>
          <div
            class="form-group mb-2"
          >
            <input
              class="form-control"
              name="registerPassword2"
              placeholder="Repeat the password"
              type="password"
              value=""
            />
          </div>
          <div
            class="d-grid gap-2"
          >
            <input
              class="btnSubmit"
              type="submit"
              value="Create account"
            />
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;

exports[`Tests en <AppRouter /> must display login in case of not being authenticated 1`] = `
<div>
  <div
    class="container login-container"
  >
    <div
      class="row"
    >
      <div
        class="col-md-6 login-form-1"
      >
        <h3>
          Entry
        </h3>
        <form>
          <div
            class="form-group mb-2"
          >
            <input
              class="form-control"
              name="loginEmail"
              placeholder="Email"
              type="text"
              value=""
            />
          </div>
          <div
            class="form-group mb-2"
          >
            <input
              class="form-control"
              name="loginPassword"
              placeholder="Password"
              type="password"
              value=""
            />
          </div>
          <div
            class="d-grid gap-2"
          >
            <input
              class="btnSubmit"
              type="submit"
              value="Login"
            />
          </div>
        </form>
      </div>
      <div
        class="col-md-6 login-form-2"
      >
        <h3>
          Register
        </h3>
        <form>
          <div
            class="form-group mb-2"
          >
            <input
              class="form-control"
              name="registerName"
              placeholder="Name"
              type="text"
              value=""
            />
          </div>
          <div
            class="form-group mb-2"
          >
            <input
              class="form-control"
              name="registerEmail"
              placeholder="Email"
              type="email"
              value=""
            />
          </div>
          <div
            class="form-group mb-2"
          >
            <input
              class="form-control"
              name="registerPassword"
              placeholder="Password"
              type="password"
              value=""
            />
          </div>
          <div
            class="form-group mb-2"
          >
            <input
              class="form-control"
              name="registerPassword2"
              placeholder="Repeat the password"
              type="password"
              value=""
            />
          </div>
          <div
            class="d-grid gap-2"
          >
            <input
              class="btnSubmit"
              type="submit"
              value="Create account"
            />
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;

exports[`Tests en <AppRouter /> must show the calendar if we are authenticated 1`] = `
<div>
  <div
    class="container login-container"
  >
    <div
      class="row"
    >
      <div
        class="col-md-6 login-form-1"
      >
        <h3>
          Entry
        </h3>
        <form>
          <div
            class="form-group mb-2"
          >
            <input
              class="form-control"
              name="loginEmail"
              placeholder="Email"
              type="text"
              value=""
            />
          </div>
          <div
            class="form-group mb-2"
          >
            <input
              class="form-control"
              name="loginPassword"
              placeholder="Password"
              type="password"
              value=""
            />
          </div>
          <div
            class="d-grid gap-2"
          >
            <input
              class="btnSubmit"
              type="submit"
              value="Login"
            />
          </div>
        </form>
      </div>
      <div
        class="col-md-6 login-form-2"
      >
        <h3>
          Register
        </h3>
        <form>
          <div
            class="form-group mb-2"
          >
            <input
              class="form-control"
              name="registerName"
              placeholder="Name"
              type="text"
              value=""
            />
          </div>
          <div
            class="form-group mb-2"
          >
            <input
              class="form-control"
              name="registerEmail"
              placeholder="Email"
              type="email"
              value=""
            />
          </div>
          <div
            class="form-group mb-2"
          >
            <input
              class="form-control"
              name="registerPassword"
              placeholder="Password"
              type="password"
              value=""
            />
          </div>
          <div
            class="form-group mb-2"
          >
            <input
              class="form-control"
              name="registerPassword2"
              placeholder="Repeat the password"
              type="password"
              value=""
            />
          </div>
          <div
            class="d-grid gap-2"
          >
            <input
              class="btnSubmit"
              type="submit"
              value="Create account"
            />
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;
